### PR TITLE
Add force_write flag to save_batch

### PR DESCRIPTION
**Pull request by dcbrandao**
@akefeli wanted to be able to force a write to database regardless of
whether the database has a record with the same primary key, so I've
added the force_write flag to the save_batch method to enable this.
I also started adding testing for some of the model methods, and a
couple of bugs I found in the process were fixed.

### CODE REVIEW

File: spanner_orm/tests/model_api_test.py [hunk 1]
```diff
@@ -49,3 +50,53 @@ def test_find_result(self, find):
     self.assertEqual(result.key, 'key')
     self.assertEqual(result.value_1, 'value_1')
     self.assertIsNone(result.value_2)
+
+  @mock.patch('spanner_orm.api.SpannerApi.insert')
+  def test_create_calls_api(self, insert):
+    mock_transaction = mock.Mock()
+    models.SmallTestModel.create(mock_transaction, key='key', value_1='value')
+
+    insert.assert_called_once()
```
Comment (akefeli) [inline spanner_orm/tests/model_api_test.py]: this could use `assert_api_called` helper-

File: spanner_orm/tests/model_api_test.py [hunk 2]
```diff
@@ -49,3 +50,53 @@ def test_find_result(self, find):
...
+  def assert_api_called(self, mock_api, mock_transaction):
+    mock_api.assert_called_once()
+    (transaction, table, columns, values), _ = mock_api.call_args
+    self.assertEqual(transaction, mock_transaction)
+    self.assertEqual(table, models.SmallTestModel.table)
+    self.assertEqual(list(columns), ['key', 'value_1', 'value_2'])
+    self.assertEqual(list(values), [['key', 'value', None]])
+
+  @mock.patch('spanner_orm.api.SpannerApi.insert')
+  def test_save_batch_inserts(self, insert):
```
Comment (akefeli) [inline spanner_orm/tests/model_api_test.py]: this method and `test_create_calls_api` seem to have same model, yet former asserts with `value_2` column, latter doesn't. I don't see `value_2` anywhere in this method, how does it assert that column?

Comment (dcbrandao) [reply spanner_orm/tests/model_api_test.py]: batch_save() always sets all columns on a model, create() only sets columns that were explicitly passed in

File: spanner_orm/tests/model_api_test.py (hunk 1 above)
Comment (dcbrandao) [reply spanner_orm/tests/model_api_test.py]: that helper is just for batch_save calls, for the reason mentioned below

**Pull request merged**